.github/workflows/rust.yml

```
@@ -6,24 +6,47 @@ on:
   schedule:
     - cron: '0 0 1 * *'
 
+defaults:
+  run:
+    working-directory: fiat-rust
+
 jobs:
   test-rust:
 
     runs-on: ubuntu-latest
 
     concurrency:
-      group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+      group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-${{ github.job }}
       cancel-in-progress: true
 
     steps:
     - uses: actions/checkout@v3
     - name: cargo build
-      run: (cd fiat-rust; cargo build --verbose)
+      run: cargo build --verbose
     - name: cargo test
-      run: (cd fiat-rust; cargo test --verbose)
-    - name: cargo test (no_std)
-      run: (cd fiat-rust; cargo test --no-default-features --verbose)
+      run: cargo test --verbose
     - name: curve25519-dalek test
-      run: etc/ci/test-fiat-rust-curve25519-dalek.sh
+      run: (cd .. ; etc/ci/test-fiat-rust-curve25519-dalek.sh)
     - name: crate-crypto/Ed448-Goldilocks test
-      run: etc/ci/test-fiat-rust-crate-crypto-Ed448-Goldilocks.sh
+      run: (cd .. ; ../etc/ci/test-fiat-rust-crate-crypto-Ed448-Goldilocks.sh)
+
+  build-nostd:
+    name: Build on no_std target (thumbv7em-none-eabi)
+    runs-on: ubuntu-latest
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-${{ github.job }}
```

```suggestion
      group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}-${{ job.name }}
```
I can try adding this change from the web interface